### PR TITLE
Slight profile page header refactoring

### PR DESCRIPTION
**Pull request by notbakaneko**
More header links coming soon™️?
- less duplicated code for rendering the same thing
- stops using `cover_url` on user

### CODE REVIEW

File: resources/assets/lib/globals.d.ts [hunk 1]
```diff
@@ -122,9 +122,9 @@ interface Country {
 }
 
 interface Cover {
-  custom_url?: string;
-  id?: string;
-  url?: string;
+  custom_url: string | null;
+  id: string | null;
+  url: string | null;
```
Comment (nanaya) [inline resources/assets/lib/globals.d.ts:127]: the one from backend always returns a url...

**Review by nanaya — CHANGES_REQUESTED** [review]

File: resources/assets/lib/globals.d.ts (hunk 1 above)
Comment (notbakaneko) [reply resources/assets/lib/globals.d.ts:127]: `UserNotFound` for the user card uses `null`

File: app/Transformers/ProfileBannerTransformer.php [hunk 2]
```diff
@@ -9,12 +9,8 @@
 
 class ProfileBannerTransformer extends TransformerAbstract
 {
-    public function transform(?ProfileBanner $banner)
+    public function transform(ProfileBanner $banner)
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply app/Transformers/ProfileBannerTransformer.php:12]: The docs seem correct? Unless optional doesn't mean could also be null? 🤔

[the remark this answers is not in the document]
Comment (nanaya) [reply app/Transformers/ProfileBannerTransformer.php:12]: hmm, if an endpoint says it includes "active_tournament_banner", does it mean it always contains it or may still be null? :thinking: (also the current one doesn't even say those may be null)

**Review by nanaya — CHANGES_REQUESTED** [review]

**Review by nanaya — CHANGES_REQUESTED** [review]

**Review by nanaya — APPROVED** [review]

**Pull request merged**